### ISSUE TITLE
Access command line arguments (argv and argc) in AAarch64/Darwin assembly

### ISSUE THREAD

**Issue by jrosengarden** [below/HelloSilicon#22]
I'm hoping you can possibly answer this question???

I've been working thru the book and your code changes in order for everything to work on the Machine We Must Not Speak About.

I'm trying to extend one of the Chapter 4 exercises and access the command line argument(s) (argc & argv) within my assembly code.  I've searched high and low and just can't find anything that applies (or works).

Could you possibly provide any clues/hints on how to access the command line arguments (argv and argc) within the assembly program.  

This very simple program is working fine but if you read the TODO, in the attached code,  what I'm trying to do should make sense.

[case.txt](https://github.com/below/HelloSilicon/files/5138412/case.txt)

Thanks much!

**Comment by below** [comment]
Gladly, it is quite easy. This is the entry function:
```
int main (int argc, const char * argv[])
```

While I have not figured out why this is not true for `printf`, it is true here that the arguments are in the registers:
`x0` holds `argc`
`x1` holds `argv`

Let's invoke our executable with `lldb HelloSilicon foo`, and look at the register when our `_start` routine is called:
```
(lldb) re r sp x0 x1
      sp = 0x000000016fdff970
      x0 = 0x0000000000000002
      x1 = 0x000000016fdff9a0
```
As you can see, x0 contains 2, and x1 points to a position 48 bytes above the current SP (and we can find the value for argc at 40 bytes, too).

When we examine the memory pointed to at x1, we see the elements of argv, which are in turn pointers:
```
(lldb) m read -fx -c32 0x000000016fdff9a0
0x16fdff9a0: 0x6fdffb20 0x00000001 0x6fdffb53 0x00000001
```

The memory pointed to by the doubleword at x1 contains, as expected, the full path to the executable:
```
(lldb) m read -fc -c51 0x000000016fdffb20
0x16fdffb20: /Users/below/dev/HelloSilicon/Ch
0x16fdffb40: apter 1/HelloWorld\0
```

And finally, the doubleword at [x1, #8] points to our argument, foo:
```
(lldb) m read -fc -c4 0x000000016fdffb53
0x16fdffb53: foo\0
```

With this information, it should be easy to adopt your code! Let me know if it works!

**Comment by jrosengarden** [comment]
Thank-you VERY, VERY, VERY much.  That worked perfectly and I've modified my program to now grab the command line argument(s) entered and act on them accordingly.

Again....thanks tons.  Much appreciated!!!!!!

**Comment by below** [comment]
Please see my adaption of your Code in Chapter 4/case.s and feel free to add or edit it

**Comment by jrosengarden** [comment]
Well...I'm back.....this time with a problem (self-inflicted wound?)

I've been moving merrily along through your "Hello Silicon" project...learning tons (and having a blast).
Along the way I've been keeping my "MWMNSA" up-to-date with the OS betas and the Xcode betas.

My problem started this morning when I (foolishly?) installed the publicly available version of Big Sur - 11.0.1

Now all my projects fail to compile/link successfully (actually the failure is during the link phase).
Here is a sample of the error I'm getting


jeffrosengarden@MacMiniDTK Chapter 4 % make case
ld -o case -lSystem -syslibroot `xcrun -sdk macosx --show-sdk-path` -e _start -arch arm64   case.o
ld: warning: ignoring file /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/lib/libSystem.tbd, missing required architecture arm64 in file /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/lib/libSystem.tbd
ld: dynamic main executables must link with libSystem.dylib for architecture arm64
make: *** [case] Error 1

I'm not sure what to do here.  Any help/advice would be greatly appreciated.

Thanks much!
